NBitcoin/Utils.cs

```diff
@@ -699,10 +698,39 @@ public static IPEndPoint ParseIpEndpoint(string endpoint, int defaultPort)
 			}
 			catch(FormatException)
 			{
+#if !WINDOWS_UWP
 				address = Dns.GetHostEntry(ip).AddressList[0];
+#else
+				string adr = DnsLookup(ip).Result;
+				// if not resolved behave like GetHostEntry
+				if (adr == string.Empty)
+					throw new SocketException(11001);
+				else
+					address = IPAddress.Parse(adr);
+#endif
 			}
 			return new IPEndPoint(address, port);
 		}
+
+#if WINDOWS_UWP
+		private static async Task<string> DnsLookup(string remoteHostName)
+		{
+			IReadOnlyList<EndpointPair> data = await DatagramSocket.GetEndpointPairsAsync(new HostName(remoteHostName), "0");

```

Comment:
Windows.Foundation.IAsyncOperation<IReadOnlyList<EndpointPair>> seem to have not .ConfigureAwait method. My sample app work without it. Do you think that I have to put the code in this way ?

private static async Task<string> DnsLookup(string remoteHostName)
{
SemaphoreSlim mutex = new SemaphoreSlim(1);
IReadOnlyList<EndpointPair> data;

  await mutex.WaitAsync().ConfigureAwait(false);
  try
  {
    data = await DatagramSocket.GetEndpointPairsAsync(new HostName(remoteHostName), "0");
  }
  finally
  {
    mutex.Release();
  }

[....]
}`